lib/tasks/scripts/message_patients.rb

```diff
@@ -0,0 +1,98 @@
+class MessagePatients
+  def self.call(*args)
+    new(*args).call
+  end
+
+  attr_reader :patients, :message, :channel, :dryrun, :verbose, :report
+  VALID_CHANNELS = [:whatsapp, :sms]
+
+  def initialize(patients, message, channel: :whatsapp, dryrun: false, verbose: true)
+    @patients = patients
+    @message = message
+    @channel = channel
+    @dryrun = dryrun
+    @verbose = verbose
+    @report = {}
+  end
+
+  def call
+    unless patients.is_a?(ActiveRecord::Relation)
+      raise ArgumentError, "Patients should be passed in as an ActiveRecord::Relation."
+    end
+
+    unless VALID_CHANNELS.include?(channel)
+      raise ArgumentError, "Message channels can only be of types: #{VALID_CHANNELS}"
+    end
+
+    log("Total patients that will be contacted via #{channel}: #{contactable_patients.count}.")
+
+    return if dryrun
+
+    send_messages
+    print_report
+    self
+  end
+
+  private
+
+  def send_messages
+    contactable_patients.each do |patient|
+      phone_number = patient.latest_mobile_number
+
+      begin
+        response =
+          if whatsapp?
+            NotificationService
+              .new
+              .send_whatsapp(phone_number, message)
```

Comment:
One-line em if you'd like 😄 `NotificationService.new.send_whatsapp(phone_number, message)` isn't too long.